### DISCUSSION TITLE
Issue with AzureDevops during Test task

### DISCUSSION THREAD

**Discussion by azerios** [fluentmigrator/fluentmigrator#1588] [Q&A] [answered]
**Describe the bug**
I'm using migrations inside my application, using CQRS architecture I wrote some tests about queries and commands.
Once the Test task is enable in the AzureDevops pipeline, for each tests linked to queries; all migrations are re-run so my task is always in timeout

**To Reproduce**
Create an empty project with 3 migrations, one to create users table, one migration to populate Admin and one migration to populate Testing user to use inside tests (using a specific tag)
Create one command to create a user and another one to update the login
Write a test to update a user like the Test user update the login of Admin User
Write a test to obtain all users inside the database
Write a test to obtain all admin users inside the database
When I run tests locally all tests are executed within a minute from visual studio
When I run tests form dotnet test --configuration release all tests are executed within 30 seconds

**Expected behavior**
The CI/CD process should run migrations only once and not 3 times if I have 3 tests linked to the queries
It's like my database fixture configuration is ignored or skipped

**Information (please complete the following information):**
 - OS: Windows Server 2016,
 - Platform .NET 6.0
 - FluentMigrator version : 3.3.2

**Comment by jzabroski** [comment, marked as answer]
@azerios I think this is a misconception about your overall Ci/CD process.  I don't want to engage in a thread where I'm giving out consulting advice, but you should look into global set-up/teardown in your test framework.

If you're using xunit.net, see https://xunit.net/docs/shared-context
